src/JustSaying/Messaging/Monitoring/LogOperation.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+
+namespace JustSaying.Messaging.Monitoring
+{
+    internal sealed class LogOperation : IDisposable
+    {
+        private readonly ILogger _logger;
+        private readonly string _message;
+        private readonly object[] _args;
+        private readonly Stopwatch _watch;
+
+        public LogOperation(ILogger logger, string message, params object[] args)
+        {
+            _logger = logger;
+            _message = message;
+            _args = args;
+            _watch = new Stopwatch();
+
+            _watch.Start();
+        }
+
+        public void Dispose()
+        {
+            var args = _args.Concat(new object[] { _watch.ElapsedMilliseconds }).ToArray();
```

Comment:
Should explicitly stop the stopwatch before grabbing the time.

Reply:
Done!